### DISCUSSION TITLE
zod schema not showing custom error message with number

### DISCUSSION THREAD

**Discussion by richewinqred** [react-hook-form/react-hook-form#10651] [General]
Hey, 

any ideas why I am unable to show a custom message on `number() `(`valueAsNumber: true`) inputs?

error messages are working correctly with `string()`?

```
const schema= z.object({
  companyName: z.string().min(2, { message: 'Please fill out this field' }),
  companyNumber: z.number().min(1, { message: 'Please fill out this field with numbers' }),
  email: z.string().email({ message: 'Please fill out this field' }),
  phone: z.number().min(2, { message: 'Please fill out this field with numbers' }),
});
```

```
 <TextInput
            label="Phone:
            name="phoneNumber"
            placeholder="xxxx-xxxxxxxxx"
             {...(register('phone'), { valueAsNumber: true })}
             error={errors.phone?.message} // not showing?
          />
```

**Comment by leapful** [comment]
@richewinqred 

I think it's already explained in the zod documentation, `number().min()` is a comparison of number value. You expected field value to be a number but then want to check length of characters which is impossible for a number value and only available for string value.

<img width="774" alt="Screenshot 2023-07-10 at 14 42 46" src="https://github.com/react-hook-form/react-hook-form/assets/18494222/8f2d0e3e-5ada-4892-8c7e-6f3e378d2439">
